File: Computer Programming/C++/Hashing/Find Common Elements Between Two Arrays.cpp
class Solution {
public:
    vector<int> findIntersectionValues(vector<int>& nums1, vector<int>& nums2) 
    {
        int n1 = nums1.size();
        int n2 = nums2.size();

        unordered_map<int,int> hash1;
        unordered_map<int,int> hash2;

        int ans1 = 0;
        int ans2 = 0;
        for(auto it : nums1)
        {
            hash1[it] ++;
        }     
        for(auto it : nums2)
        {
            hash2[it] ++;
        }

        for(auto it : nums2)
        {
            if(hash1[it] > 0)
            {
                ans2++;
            }
        }

        for(auto it : nums1)
        {
            if(hash2[it] > 0)
            {
                ans1++;
            }
        }

        return {ans1,ans2};


    }
};

File: Computer Programming/Python/Hashing/Find Common Elements Between Two Arrays.py
class Solution(object):
    def findIntersectionValues(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        
        mp1 = {}
        mp2 = {}

        n = len(nums1)
        m = len(nums2)

        for i in range(n):
            if nums1[i] in mp1:
                mp1[nums1[i]] += 1
            else:
                mp1[nums1[i]] = 1

        
        for i in range(m):
            if nums2[i] in mp2:
                mp2[nums2[i]] += 1

            else:
                mp2[nums2[i]] = 1

        
        count1 = 0
        count2 = 0

        for i in range(m):
            if nums2[i] in mp1:
                count2 += 1

        for i in range(n):
            if nums1[i] in mp2:
                count1 += 1

        return [count1,count2]
